lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,198 @@
+
+	b := buildDataParams{
+		target:     params.Target,
+		thumbprint: params.Thumbprint,
+		datacenter: &params.Datacenter,
+	}
+
+	d, err := buildData(op, b, principal)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	d.ID = params.VchID
+
+	helper, err := getDatastoreHelper(op, d)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	logFilePaths, err := getAllLogFilePaths(op, helper)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	output, err := getContentFromLogFiles(op, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(err.Error())
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogOK().WithPayload(output)
+}
+
+// getDatastoreHelper validates the VCH and returns the datastore helper for the VCH. It errors when validation fails or when datastore is not ready
+func getDatastoreHelper(op trace.Operation, d *data.Data) (*datastore.Helper, error) {
+	// TODO (angiew): abstract some of the boilerplate into helpers in common.go
+	validator, err := validateTarget(op, d)
+	if err != nil {
+		return nil, util.WrapError(http.StatusBadRequest, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(http.StatusNotFound, fmt.Sprintf("Unable to find VCH %s: %s", d.ID, err))
+	}
+
+	if err := validate.SetDataFromVM(validator.Context, validator.Session.Finder, vch, d); err != nil {
+		return nil, util.NewError(http.StatusInternalServerError, fmt.Sprintf("Failed to load VCH data: %s", err))
+	}
+
+	// Get VCH configuration
+	vchConfig, err := executor.GetNoSecretVCHConfig(vch)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve VCH information: %s", err)
+	}
+
+	// Relative path of datastore folder
+	vmPath := vchConfig.ImageStores[0].Path
+
+	// Get VCH datastore object
+	ds, err := validator.Session.Finder.Datastore(validator.Context, vchConfig.ImageStores[0].Host)
+	if err != nil {
+		return nil, util.NewError(http.StatusNotFound, fmt.Sprintf("Datastore folder not found for VCH %s: %s", d.ID, err))
+	}
+
+	// Create a new datastore helper for file finding
+	helper, err := datastore.NewHelper(op, validator.Session, ds, vmPath)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get datastore helper: %s", err)
+	}
+
+	return helper, nil
+}
+
+// getAllLogFilePaths returns a list of all log file paths under datastore folder, errors out when no log file found
+func getAllLogFilePaths(op trace.Operation, helper *datastore.Helper) ([]string, error) {
+	res, err := helper.Ls(op, "")
```

Comment:
If you look at the implementation of `helper.Ls` it would be trivial to change it so that you can actually provide a pattern instead of a static name. There is already a MatchPattern defined as part of the search spec.
That would basically eliminate the need for this function.

Reply:
Filed #6733 for this.